### PR TITLE
fix the npm ci

### PR DESCRIPTION
**Pull request by vihoyau**
1.new repo PR to change the npm ci

### CODE REVIEW

**Comment by wu-sheng** [comment]
You need to describe why you want to change this.

**Comment by vihoyau** [comment]
fix the npm ci reason:
1.npm ci have a command "npm install", it don't effective about webpack.
2.we can use the command "yarn install", it can install webpack node_moudle and so on.

**Review by Fine0830 — COMMENTED** [review]
If we want to use `yarn` replace `npm`, we need to replace all of script commands and `lock.json` file, not just `npm install`.

**Comment by vihoyau** [comment]
> If we want to use `yarn` replace `npm`, we need to replace all of script commands and `lock.json` file, not just `npm install`.

can i  need to replace all ? it’s difficult

**Comment by vihoyau** [comment]
> If we want to use `yarn` replace `npm`, we need to replace all of script commands and `lock.json` file, not just `npm install`.
thank you , i have done

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
   },
   "scripts": {
     "build": "cross-env NODE_ENV=production webpack",
-    "rebuild": "rm -rf ./node_modules && rm -rf ./lib && npm install && npm run build",
+    "rebuild": "rm -rf ./node_modules && rm -rf ./lib && yarn install && npm run build",
```
Comment (Fine0830) [inline package.json:32]:
```suggestion
    "rebuild": "rm -rf ./node_modules && rm -rf ./lib && yarn install && yarn run build",
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -34,12 +34,11 @@ jobs:
...
         node-version: ${{ matrix.node-version }}
-    - name: npm install and build
+    - name: yarn install and build
       run: |
-        npm ci
-        npm run build --if-present
-      env:
-        CI: true
+        yarn install
+        yarn webpack
```
Comment (Fine0830) [inline .github/workflows/build.yml:40]: Why do you change this?

**Review by Fine0830 — COMMENTED** [review]
Please remove `package-lock.json`.

**Comment by wu-sheng** [comment]
As npm to yarn, is doc required to update?

**Comment by Fine0830** [comment]
> As npm to yarn, is doc required to update?

Yes, please change `README.md` and `release.md`. @vihoyau

File: README.md [hunk 3]
```diff
@@ -12,10 +12,10 @@ NOTICE,  SkyWalking Client JS 0.8.0 and later versions require SkyWalking v9.
 # Usage
 
 ## Install  
-The `skywalking-client-js` runtime library is available at [npm](https://www.npmjs.com/package/skywalking-client-js).
+The `skywalking-client-js` runtime library is available at [yarn]([https://classic.yarnpkg.com/en/package/webpack/skywalking-client-js)].
```
Comment (wu-sheng) [inline README.md:15]: Which yarn version do we need?

**Review by wu-sheng — COMMENTED** [review]
@kezhenxu94 Does skywalking-eyes support yarn dependencies analysis? If not, is it easy to add?

File: README.md (hunk 3 above)
Comment (vihoyau) [reply README.md]: in my local, i use version is 1.22.11 , it can effective for webpack

Comment (wu-sheng) [reply README.md]: This is a doc for end users and developers, it should be very clear which versions(such as 1.22.x+ or something).

Comment (vihoyau) [reply README.md]: Thanks, Let me add it.

**Review by kezhenxu94 — CHANGES_REQUESTED** [review]
The `yarn.lock` file seems to be broken, are you sure the comment style in `yarn.lock` is `/* */`? I run `yarn install` and it failed. Please confirm.

Also, I'm wondering whether we need license header in such file, it's totally generated

**Comment by vihoyau** [comment]
How to bypass license detection for the yarn.lock？

**Comment by vihoyau** [comment]
> How to bypass license detection for the yarn.lock？
thanks, i have done

File: README.md [hunk 4]
```diff
@@ -12,10 +12,11 @@ NOTICE,  SkyWalking Client JS 0.8.0 and later versions require SkyWalking v9.
 # Usage
 
 ## Install  
-The `skywalking-client-js` runtime library is available at [npm](https://www.npmjs.com/package/skywalking-client-js).
+The `skywalking-client-js` runtime library is available at [yarn](https://classic.yarnpkg.com/en/package/skywalking-client-js).
 
+The yarn Version is V1.22.11 Or Higher. [Yarn Version](https://github.com/yarnpkg/yarn/releases)
 ```
-npm install skywalking-client-js --save
+yarn install skywalking-client-js --save
```
Comment (Fine0830) [inline README.md:19]: The `skywalking-client-js` runtime library is available at [npm](https://www.npmjs.com/package/skywalking-client-js).

```
npm install skywalking-client-js --save
```
```
yarn add skywalking-client-js
```

File: release.md [hunk 5]
```diff
@@ -142,18 +142,18 @@ SkyWalking Resources:
 - Apache SkyWalking Team
 ```
 
-6. publish package to NPM
+6. publish package to YARN
```
Comment (Fine0830) [inline release.md:145]: Here let's keep `npm`.

File: README.md [hunk 6]
```diff
@@ -12,10 +12,11 @@ NOTICE,  SkyWalking Client JS 0.8.0 and later versions require SkyWalking v9.
 # Usage
 
 ## Install  
-The `skywalking-client-js` runtime library is available at [npm](https://www.npmjs.com/package/skywalking-client-js).
+The `skywalking-client-js` runtime library is available at [yarn](https://classic.yarnpkg.com/en/package/skywalking-client-js).
```
Comment (Fine0830) [inline README.md:15]:
```suggestion
The skywalking-client-js runtime library is available at [npm](https://www.npmjs.com/package/skywalking-client-js).
```

File: README.md [hunk 7]
```diff
@@ -12,10 +12,11 @@ NOTICE,  SkyWalking Client JS 0.8.0 and later versions require SkyWalking v9.
...
 
+The yarn Version is V1.22.11 Or Higher. [Yarn Version](https://github.com/yarnpkg/yarn/releases)
 ```
-npm install skywalking-client-js --save
+yarn add skywalking-client-js --save
```
Comment (Fine0830) [inline README.md:19]: Change `yarn add skywalking-client-js --save` to `yarn add skywalking-client-js`.
And add this to here.
```
npm install skywalking-client-js --save
```

File: README.md [hunk 8]
```diff
@@ -12,9 +12,11 @@ NOTICE,  SkyWalking Client JS 0.8.0 and later versions require SkyWalking v9.
 # Usage
 
 ## Install  
-The `skywalking-client-js` runtime library is available at [npm](https://www.npmjs.com/package/skywalking-client-js).
+The `skywalking-client-js` runtime library is available at [yarn](https://classic.yarnpkg.com/en/package/skywalking-client-js).
 
+The yarn Version is V1.22.11 Or Higher. [Yarn Version](https://github.com/yarnpkg/yarn/releases)
 ```
+yarn add skywalking-client-js
```
Comment (Fine0830) [inline README.md:19]: Could you put it in another code segment(``` xxx ```).

File: README.md [hunk 9]
```diff
@@ -12,9 +12,11 @@ NOTICE,  SkyWalking Client JS 0.8.0 and later versions require SkyWalking v9.
 # Usage
 
 ## Install  
-The `skywalking-client-js` runtime library is available at [npm](https://www.npmjs.com/package/skywalking-client-js).
+The `skywalking-client-js` runtime library is available at [yarn](https://classic.yarnpkg.com/en/package/skywalking-client-js).
```
Comment (Fine0830) [inline README.md:15]: Here let's keep `npm`.

File: .github/workflows/build.yml [hunk 10]
```diff
@@ -34,13 +34,12 @@ jobs:
       uses: actions/setup-node@v1
       with:
         node-version: ${{ matrix.node-version }}
-    - name: npm install and build
+    - name: yarn install and build
       run: |
-        npm ci
-        npm run build --if-present
+        yarn install
```
Comment (Fine0830) [inline .github/workflows/build.yml:39]: Before this, please install `yarn` .

`npm install --global yarn`

**Comment by wu-sheng** [comment]
@vihoyau Why another account could close your PR?

**Comment by vihoyau** [comment]
it's my another developer organazition

**Comment by vihoyau** [comment]
> @vihoyau Why another account could close your PR?

how to solve this problem?

**Comment by wu-sheng** [comment]
I don't know, I can see it even remove your commit somehow.

**Comment by kezhenxu94** [comment]
@vihoyau

**Pull request closed without merging**